### ISSUE TITLE
ICE: Internal compiler error: Block while has a misplaced terminator.

### ISSUE THREAD

**Issue by Braqzen** [FuelLabs/sway#6205]
`Forc: latest (0.60.0)`

Repro: `forc build`

```sway
contract;

abi TestContract {
    fn test_method();
}

impl TestContract for Contract {
    fn test_method() {
        while (true && true) {}
    }
}
```

Context: the issue is having more than one `&&`. I've reduced this from a massive codebase. I have not attempted to try different combinations (logical or otherwise)

**Comment by Braqzen** [comment]
cc @IGI-111 this is urgent for a known project integration

**Comment by ironcev** [comment]
This issue is fixed with #6146 and the code compiles with Forc 0.61.1.
